### PR TITLE
Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by estesp**
Since the `Vagrantfile` was also being used by Travis this PR removes Travis at the same time as adding Actions, given Travis would simply fail given the changes to the Vagrantfile which are aligned with using the MacOS instance in GH Actions.

You can see a successful run of these changes in my fork [here](https://github.com/estesp/cgroups/runs/751664439?check_suite_focus=true)

Fixes: #164 (which was a result of the matrix setup in Travis overriding the standard script section--GH Actions allows us to have a common project check area for DCO and headers, and then run 2 dependent jobs after that for v1 and v2 respectively)

Signed-off-by: Phil Estes <estesp@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by estesp** [comment]
Note that no CI will run on this PR because Travis's config file is removed, and actions only run in the fork, so until this merged, PRs won't run the actions defined here within the PR as a security measure. See the link above to see the output from these exact changes in my fork.

File: Vagrantfile [hunk 1]
```diff
@@ -13,13 +13,34 @@ Vagrant.configure("2") do |config|
...
-config exclude kernel,kernel-core
-config install_weak_deps false
-update
-install git golang-go
-ts run
+    set -eux -o pipefail
+    # configuration
+    GO_VERSION="1.13.11"
+
+	# install gcc and Golang
```
Comment (AkihiroSuda) [inline Vagrantfile:20]: nit: indentation

Comment (estesp) [reply Vagrantfile]: thanks! fixed

**Comment by estesp** [comment]
2 LGTMs so merging to enable full CI again in this repo. Thanks!

**Comment by estesp** [comment]
Merge to master run shows actions CI now active: https://github.com/containerd/cgroups/runs/754016151

**Pull request merged**